query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -450,7 +450,9 @@ private String convertLiteralToString(LiteralConstant literal, Params.Builder pa
     String ret = null;
     switch (value.getValueType()) {
       case STRING_ARRAY:
-        ret = buildArrayValue(value.getStringArrayList(), paramsBuilder::addStringParam);
+        List<String> values =
+            value.getStringArrayList().size() > 0 ? value.getStringArrayList() : List.of("null");
```

Comment:
Why is gateway service converting null string to empty ? Shouldn't be fix that instead of expecting every converter to handle this ?

Reply:
'null' is special case on how it is stored in Pinot and needs to be converted to an appropriate value. Isn't it?